gufe/chemicalsystem.py

```diff
@@ -126,6 +126,54 @@ def total_charge(self):
                 total_charge += fc
         return total_charge
 
+    def contains(self, item: Component | type[Component]) -> bool:
+        """Check if a Component or Component type is in this ChemicalSystem.
+
+        Parameters
+        ----------
+        item : Component or type of Component
+            The Component instance or class to check for.
+
+        Returns
+        -------
+        bool
+            Returns True if the item is found in the ChemicalSystem, False otherwise.
+        """
+        # check the input types
+        if not (isinstance(item, Component) or (isinstance(item, type) and issubclass(item, Component))):
```

Comment:
Tests still pass if you remove the `(isinstance(item, type)` check, and off the top of my head I can't think of an edge case where you would need it. Could you either remove this or add a test for an applicable edge case?
```suggestion
        if not (isinstance(item, Component) or issubclass(item, Component))):
```